src/pages/DirectMessage.tsx

```diff
@@ -108,25 +119,55 @@ const DirectMessage = () => {
     (o) => o.otherUserId === Number(userId2)
   );
 
-  useEffect(() => {
-    const fetchDirectMessage = async () => {
-      const result = await axios.get(
-        `http://localhost:3001/api/messages/${userId1}/${userId2}`
+  const fetchUser = async () => {
+    try {
+      const otherUser = await axios.get(
+        `http://localhost:3001/api/messages/${userId2}`
       );
-      setMessages(result.data.messages as Message[]);
+
       dispatch(
         setSelectedConversation({
-          ...result.data.otherUser,
+          ...otherUser.data,
           userId: Number(userId2),
         })
       );
-    };
-    fetchDirectMessage();
-  }, [dispatch, userId1, userId2]);
+    } catch (error) {
+      console.log("error fetching user data:", error);
+    }
+  };
+
+  const fetchDirectMessage = async ({ pageParam = 1 }) => {
+    try {
+      const result = await axios.get(
+        `http://localhost:3001/api/messages/${userId1}/${userId2}?offset=${pageParam}`
+      );
+
+      if (pageParam > 1) {
+        setMessages([...messages, ...result.data] as Message[]);
+      } else {
+        setMessages(result.data as Message[]);
+      }
+
+      return result.data;
+    } catch (error) {
+      console.log("error fetching posts:", error);
+    }
+  };
+
+  const { error, status, hasNextPage, fetchNextPage } = useInfiniteQuery({
```

Comment:
All of this stuff seems to be repeated pretty often. It'd be nice if we could make a generic wrapper component that puts all of this infinite scroll logic in the same spot. I think we can put the `useEffect` and `useInfiniteQuery` calls that come with these infinite scroll lists into a component to avoid copy pasting this boilerplate

Reply:
Done!